core/app/services/uffizzi_core/compose_file/parsers/services/image_parser_service.rb

```diff
@@ -28,44 +27,56 @@ def parse(image)
 
     private
 
-    def valid_image_url?(image_name)
-      url = prepare_image_url(image_name)
-
-      URI(url).host.present? && URI(url).host =~ /\w+\.\w+/ && URI(url).path.present?
-    end
-
-    def prepare_image_url(image_name)
-      if image_name.start_with?('https://')
-        image_name
+    def get_image_path_and_tag(value, parse_error)
+      image_path_parts = value.downcase.split(':')
+      case image_path_parts.size
+      when 1
+        image_path_parts[0]
+      when 2
+        uri_pattern = /\A\w[\w.-]+:\d+/
+        tag_pattern = /:\w[\w.-]*\z/
+        if uri_pattern.match?(value)
+          "#{image_path_parts[0]}:#{image_path_parts[1]}"
+        elsif tag_pattern.match?(value)
+          [image_path_parts[0], image_path_parts[1]]
+        else
+          raise parse_error
+        end
+      when 3
+        ["#{image_path_parts[0]}:#{image_path_parts[1]}", image_path_parts[2]]
       else
-        "https://#{image_name}"
+        raise parse_error
       end
     end
 
-    def parse_image_url(image_name)
-      prepared_url = prepare_image_url(image_name)
-
-      uri = URI(prepared_url)
-      url = uri.host
-      path = uri.path.delete_suffix('/').delete_prefix('/')
+    def url?(image_path)
+      uri = URI(add_https_if_needed(image_path))
+    rescue URI::InvalidURIError
+      false
+    else
+      uri.host.present? && uri.host =~ /\w+\.(\w+\.)*\w+/ && uri.path.present?
+    end
```

Comment:
Do we really need to have `rescue` in the top of function?

Reply:
Are you questioning its necessity or position?
For example, `URI::InvalidURIError` is raised for `URI('https://postgres:9.6')`.

Reply:
Are you questioning its necessity or position?
For example, `URI::InvalidURIError` is raised for `URI('https://postgres:9.6')`.